File: MTH103/Variance.java
public class Variance {

    public static void main(String[] args) {
        double[][] nums = {
                { 1, (double) 1 / 6 },
                { 2, (double) 1 / 6 },
                { 3, (double) 1 / 6 },
                { 4, (double) 1 / 6 },
                { 5, (double) 1 / 6 },
                { 6, (double) 1 / 6 }
        };

        System.out.printf("Variance: %.4f\n", variance(nums));
        System.out.printf("Standard Deviation: %.4f\n", standardDev(nums));
    }

    static double sum(double[][] arr) {
        double sum = 0;
        for (int i = 0; i < arr.length; i++) {
            sum += arr[i][0];
        }
        return sum;
    }

    static double mean(double[][] arr) {
        return sum(arr) / arr.length;
    }

    static double variance(double[][] arr) {
        double result = 0;
        double mean = mean(arr);
        for (int i = 0; i < arr.length; i++) {
            result += Math.pow(arr[i][0] - mean, 2) * arr[i][1];
        }
        return result;
    }

    static double standardDev(double[][] arr) {
        return Math.sqrt(variance(arr));
    }
}

File: MTH103/variance.js
const nums = [
    { value: 1, probability: 1 / 6 },
    { value: 2, probability: 1 / 6 },
    { value: 3, probability: 1 / 6 },
    { value: 4, probability: 1 / 6 },
    { value: 5, probability: 1 / 6 },
    { value: 6, probability: 1 / 6 }
];

const sum = nums.reduce((current, next) => current + next.value, 0);
const mean = sum / nums.length;
const variance = nums
    .map(obj => Math.pow(obj.value - mean, 2) * obj.probability)
    .reduce((current, next) => current + next, 0);
    
const standardDev = Math.sqrt(variance);

console.log(`Variance: ${variance.toFixed(4)}`);
console.log(`Standard Deviation: ${standardDev.toFixed(4)}`);